src/midgard/tiles.cc

```diff
@@ -51,14 +52,18 @@ void bresenham_line(double x0,
 template <class coord_t> struct closest_first_generator_t {
   valhalla::midgard::Tiles<coord_t> tiles;
   coord_t seed;
-  std::unordered_set<int32_t> queued;
+  robin_hood::unordered_set<int32_t> queued;
   int32_t subcols, subrows;
   using best_t = std::pair<double, int32_t>;
-  std::set<best_t, std::function<bool(const best_t&, const best_t&)>> queue;
+  std::priority_queue<best_t, std::vector<best_t>, std::function<bool(const best_t&, const best_t&)>>
+      queue;
+
+  // re-usable, pre-allocated vector used by dist
+  std::vector<coord_t> corners;
 
   closest_first_generator_t(const valhalla::midgard::Tiles<coord_t>& tiles, const coord_t& seed)
       : tiles(tiles), seed(seed), queued(100), queue([](const best_t& a, const best_t& b) {
-          return a.first == b.first ? a.second < b.second : a.first < b.first;
+          return a.first == b.first ? b.second < a.second : b.first < a.first;
```

Comment:
Sort order is inverted from before - the `std::priority_queue`'s `top()` function returns the "largest item", ergo we need to sort in reverse (smallest distance has highest priority).